Flaky TestAsyncRuleEvaluation

Failed here: https://github.com/prometheus/prometheus/actions/runs/7797756253/job/21265037848?pr=13545

```
=== RUN   TestAsyncRuleEvaluation/asynchronous_evaluation_of_all_independent_rules,_insufficient_concurrency
    manager_test.go:1961: 
        	Error Trace:	D:/a/prometheus/prometheus/rules/manager_test.go:1961
        	Error:      	"0.0655291" is not less than "0.06"
        	Test:       	TestAsyncRuleEvaluation/asynchronous_evaluation_of_all_independent_rules,_insufficient_concurrency
--- FAIL: TestAsyncRuleEvaluation (0.18s)
    --- FAIL: TestAsyncRuleEvaluation/asynchronous_evaluation_of_all_independent_rules,_insufficient_concurrency (0.07s)
```

Passed on re-run: https://github.com/prometheus/prometheus/actions/runs/7797756253/job/21265977651?pr=13545

Also failed here: https://github.com/prometheus/prometheus/actions/runs/7725797136/job/21060788152?pr=13197

I just got this on my desktop Linux VM:

```
$ go test -tags stringlabels ./rules/
    --- FAIL: TestAsyncRuleEvaluation/asynchronous_evaluation_with_independent_and_dependent_rules (0.06s)
        manager_test.go:2391: 
                Error Trace:    /home/vagrant/src/github.com/grafana/mimir-prometheus/rules/manager_test.go:2391
                Error:          "0.056689165" is not less than "0.04"
                Test:           TestAsyncRuleEvaluation/asynchronous_evaluation_with_independent_and_dependent_rules
```

--- a/rules/manager_test.go
+++ b/rules/manager_test.go
@@ -2043,7 +2043,7 @@ func TestBoundedRuleEvalConcurrency(t *testing.T) {
 	require.EqualValues(t, maxInflight.Load(), int32(maxConcurrency)+int32(groupCount))
 }
 
-const artificialDelay = 10 * time.Millisecond
+const artificialDelay = 15 * time.Millisecond
 
 func optsFactory(storage storage.Storage, maxInflight, inflightQueries *atomic.Int32, maxConcurrent int64) *ManagerOptions {
 	var inflightMu sync.Mutex

